sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -706,7 +722,9 @@ private Query computeQuery()
   @Nullable
   public TimeseriesQuery toTimeseriesQuery()
   {
-    if (grouping == null || grouping.getHavingFilter() != null) {
+    if (grouping == null
+        || grouping.getSubtotals().hasEffect(grouping.getDimensionSpecs())
+        || grouping.getHavingFilter() != null) {
```

Comment:
nit: it's cheaper to check if `havingFilter != null` before checking `hasEffect`

Reply:
It's probably going to be an exceedingly small effect, I'd like to leave it this way if you don't mind.

Reply:
👍